src/tools.cpp

```diff
@@ -106,9 +106,13 @@ std::tuple<char, std::string> zim::parseLongPath(const std::string& longPath)
 {
   /* Index of the namespace char; discard '/' from absolute paths */
   const unsigned int i = (longPath[0] == '/') ? 1 : 0;
-  if (i + 2 >= longPath.size() || longPath[i] == '/' || longPath[i+1] != '/')
+  if (i + 1 > longPath.size() || longPath[i] == '/' || (i + 2 <= longPath.size() && longPath[i+1] != '/'))
     throw std::runtime_error("Cannot parse path");
 
+  /* Handle namespace longPath such as "A" "/A" "A/" "/A/" */
+  if (i + 2 >= longPath.size())     
+    return std::make_tuple(longPath[i], "");
+  
   auto ns = longPath[i];
   auto shortPath = longPath.substr(i+2, std::string::npos);
 
```

Comment:
In general, it is a good practice to have only a single `return` statement in a function. In this case, it is easily achieved by slightly changing how `shortPath` is extracted from `longPath`:

```c++
  auto shortPath = longPath.substr(std::min(i+2, longPath.size()));
```

Reply:
Yes, this looks much neater. Thanks!